app/index.html

```diff
@@ -24,6 +24,8 @@
   <script type="text/javascript" src="js/livebus/livebus.factory.js"></script>
 
   <script type="text/javascript" src="js/shared/shared.module.js"></script>
+  <script type="text/javascript" src="js/shared/ucBackground/ucBackground.module.js"></script>
+  <script type="text/javascript" src="js/shared/ucBackground/ucBackground.directive.js"></script>
```

Comment:
Oh, well I can restrict that for the directive to work only with attributes :)

So are you saying that you can or can't name the files with the directive suffix, as you have written below with `validate.directive.ts` (which is for Angular 2)?

The filename is an non issue I think, the filenames can't cause any collisions.

My mistake on using the word "class"! I didn't think it was relevant in this conversation.